File: frameworks/domain.core/src/main/java/com/example/frameworks/domain/core/GlobalClockResetter.java
package com.example.frameworks.domain.core;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

import java.time.Instant;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class GlobalClockResetter {

    public static void fixAt(Instant instant) {
        GlobalClock.fixedAt(instant);
    }

    public static void reset() {
        GlobalClock.reset();
    }

}

File: frameworks/domain.core/src/main/kotlin/com/qomolangma/frameworks/domain/core/GlobalClockResetter.kt
package com.qomolangma.frameworks.domain.core

import com.qomolangma.frameworks.domain.core.GlobalClock.fixedAt
import lombok.AccessLevel
import lombok.NoArgsConstructor
import java.time.Instant

@NoArgsConstructor(access = AccessLevel.PRIVATE)
object GlobalClockResetter {
    @JvmStatic
    fun fixAt(instant: Instant?) {
        fixedAt(instant)
    }

    @JvmStatic
    fun reset() {
        GlobalClock.reset()
    }
}
